src/System.Drawing.Common/src/System/Drawing/GdiplusNative.Unix.cs

```diff
@@ -21,24 +22,45 @@ internal unsafe partial class Gdip
 
             private static IntPtr LoadNativeLibrary()
             {
+                string libraryName = RuntimeInformation.IsOSPlatform(OSPlatform.OSX) ? "libgdiplus.dylib" : "libgdiplus.so";
```

Comment:
I don't think we need this extra variable as it adds an extra check to `RuntimeInformation.IsOSPlatform(OSPlatform.OSX)` and we are already doing it later in this method to load the library.

Reply:
Yes, but we need to know whether we're looking for a `.dylib` or `.so` file in the loop that iterates over all package paths. Do you want to cache the result of IsOSPlatform in a variable instead?

Reply:
You can assign the libraryName in the `if/else` below.

Reply:
I got ya, I miss the iteration path where you're also using this variable :)

Maybe caching it or assigning libraryName inside the if and else. 

something like:

```cs
string libraryName;
if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
{
   libraryName = "libgdiplus.dylib";
   ...
}
else
{
   libraryName = "libgdiplus.so";
   ...
}

Reply:
@jkotas & @safern : great minds think alike I guess 😄 

Fixed it.